torch/csrc/api/include/torch/nn/modules/dropout.h

```diff
@@ -14,66 +14,90 @@ namespace torch {
 namespace nn {
 
 namespace detail {
+
 template <typename Derived>
-class DropoutImplBase : public torch::nn::Cloneable<Derived> {
+class _DropoutNd : public torch::nn::Cloneable<Derived> {
  public:
-  explicit DropoutImplBase(const DropoutOptions& options_ = DropoutOptions());
+  _DropoutNd(double p) : _DropoutNd(DropoutOptions().p(p)) {};
+
+  explicit _DropoutNd(const DropoutOptions& options_ = {})
+    : options(options_) {
+    TORCH_CHECK(
+        options.p() >= 0. && options.p() <= 1.,
+        "dropout probability has to be between 0 and 1, but got ",
+        options.p());
```

Comment:
We can probably move the check to the `reset()` function and call `reset()` from the constructor, which would be more consistent with the design of the other C++ modules :D